modules/core/include/opencv2/core/base.hpp

```diff
@@ -359,6 +377,18 @@ enum BorderTypes {
 #define CV_RETHROW() throw
 #endif
 
+#define CV_STR_HELPER(x) #x
+#define CV_STR(x) CV_STR_HELPER(x)
+
+#ifdef __GNUC__
+#define CV_COMPILE_TIME_WARNING(msg) CV_DO_PRAGMA(message(msg)) //TODO: use #warning(msg)
```

Comment:
Use function attributes instead:
- GCC `__attribute__ ((deprecated))`: https://gcc.gnu.org/onlinedocs/gcc-5.4.0/gcc/Function-Attributes.html
- MSVC has https://msdn.microsoft.com/en-us/library/044swk7y.aspx

Both support text messages.

Reply:
I plan to utilize this once the current compile issues go through